Assets/Scripts/ThrowController.cs

```diff
@@ -0,0 +1,216 @@
+    public delegate void OnThrowableResetEventHandler();
+    public event OnThrowableResetEventHandler OnThrowableReset;
+
+    public delegate void OnInteractEventHandler(DialogueInteractable source);
+    public event OnInteractEventHandler OnInteractBegin;
+
+    private void Start()
+    {
+        launchArcRenderer = GetComponentInChildren<LaunchArcRenderer>();
+        launchArcRenderer.gameObject.SetActive(false);  // initially hide arc
+        currentThrowables = new List<GameObject>();
+        
+        mouseHitPlane = new Plane(Vector3.up, Vector3.zero);
+        mainCamera = Camera.main;
+
+        UIManager.Instance.staminaBar.OnChange += UpdateThrowVelocity;
+    }
+
+    private void Update()
+    {
+        HandleThrowInput();
+    }
+
+    private void UpdateThrowVelocity(float fillAmount)
+    {
+        launchArcRenderer.velocity = Mathf.Lerp(minThrowVelocity, maxThrowVelocity, fillAmount / StaminaBar.FILL_MAX);
+    }
+
+    public Vector3 GetMousePosition()
+    {
+        if (mainCameraActive)
+        {
+            Vector3 position = transform.position;
+            
+            // Handle controller input
+            if (GameManager.Instance.GetPlayerController().controllerConnected && 
+                !Mathf.Approximately(Input.GetAxis("Horizontal Right Stick") + Input.GetAxis("Vertical Right Stick"),
+                    0f))
+            {
+                float horizontal = 10 * sensitivityController * Input.GetAxis("Vertical Right Stick");
+                float vertical = 10 * sensitivityController * Input.GetAxis("Horizontal Right Stick");
+                
+                return new Vector3(position.x + horizontal, position.y, position.z + vertical);
+            }
+            
+            // Handle mouse + keyboard input
+            Ray ray = mainCamera.ScreenPointToRay(Input.mousePosition);
+                    
+            if(mouseHitPlane.Raycast(ray, out float enter))
+            {
+                Vector3 hitPoint = ray.GetPoint(enter);
+                
+                float horizontal = sensitivityMouse * (hitPoint.x - position.x);
+                float vertical = sensitivityMouse * (hitPoint.z - position.z);
+            
+                return new Vector3(position.x + horizontal, position.y, position.z + vertical);
+            }
+        }
+
+        if (mainCamera != null) mainCameraActive = true;
+        return transform.position;
+    }
+
+    private void HandleThrowInput()
+    {
+        if (!GameManager.Instance.GetPlayerController().EnablePlayerInput || currentThrowables.Count <= 0) return;
+
+        // Handle controller input
+        if (GameManager.Instance.GetPlayerController().controllerConnected)
+        {
+            bool isTriggerDown = !Mathf.Approximately(Input.GetAxis(Constants.INPUT_THROW_GETDOWN), 0f);
+
+            // Right joystick is being used
+            if (!Mathf.Approximately(Input.GetAxis("Horizontal Right Stick") + Input.GetAxis("Vertical Right Stick"),
```

Comment:
This check seems to be used a couple times (also used above). Maybe make this into a function?